napari/layers/_tests/test_layer_actions.py

```diff
@@ -37,13 +38,25 @@ def _dummy():
 )
 def test_projections(mode):
     ll = LayerList()
-    ll.append(Image(np.random.rand(8, 8, 8)))
+    ll.append(Image(np.random.rand(6, 7, 8)))
     assert len(ll) == 1
     assert ll[-1].data.ndim == 3
-    _project(ll, mode=mode)
+    dims = Dims(ndim=ll[-1].data.ndim, order=(0, 1, 2))
```

Comment:
We should add another test where there are two images with mixed dimensionality (e.g. one 3D and one 4D) because I think the current approach might break then because `ViewerModel.Dims` describes the dimensions and order across all layers.

E.g. if you add a 4D image first

```python
ll.append(Image(np.random.rand(5, 6, 7, 8)))
```

but still project the 3D one, then using `axis = dims.order[axis]` is problematic because `dims.order` is some permutation of `(0, 1, 2, 3)` which means that `axis` could be out of bounds.

We solve that global/layer-specific dims correspondence problem in a few different parts of the code in unfortunately slightly different ways. I think you may be to use `Layer._world_to_layer_dims(world_dims=dims.order, ndim_world=ndim)` but that's unfortunately private and I really don't want to make it public...

Reply:
I will test this later today, indeed you might be right. I will have a look.

Reply:
It is indeed the case. What are your concerns for making it public?

One concern that I have is how easi it is to understand the API when dealing with mixed dimensionality. I think it would be better to have axis being none in which case the logic would be that napari creates a max projection which is perpendicular to the view. Otherwise create a max projection using the axis specified by the user in which case it will correspond to the layer.data axis. What do you think?

Reply:
Sorry for the slow response. What do you mean by *perpendicular to the view*? Project/integrate over all the non-visible dimensions/axes?

Reply:
Also down to find some time to code-pair on this to try to take it over the line if you're interested and have the time.

Reply:
I will work on this tomorrow morning. I did not take into account higher dimensions and need to cover this.  

What I mean with perpendicular is that the projection gets rotated 90 degrees compared to the data from which the projection is created. Lets say for illustration purposes you have  zyx data and want to do a max projection using z, instead of showing the projection as yx it shows xy. I will look a bit more in what change axis order does exactly but I know it does not change the layer data axis order.

I will keep you posted and if it doesn't work out a paired coding session would be much appreciated:)

Reply:
Ok lets do a paired coding session:) Also to discuss a bit of what should be expected behaviour here.